pkg/steps/cluster_claim_test.go

```diff
@@ -0,0 +1,231 @@
+package steps
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	prowv1 "k8s.io/test-infra/prow/apis/prowjobs/v1"
+	"k8s.io/test-infra/prow/pod-utils/downwardapi"
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+	fakectrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+
+	"github.com/openshift/ci-tools/pkg/api"
+	"github.com/openshift/ci-tools/pkg/steps/loggingclient"
+	"github.com/openshift/ci-tools/pkg/testhelper"
+	fakewatchingclient "github.com/openshift/ci-tools/pkg/util/watchingclient/fake"
+)
+
+var (
+	aClusterPool = &hivev1.ClusterPool{
```

Comment:
please use constructor funcs to avoid accidentally sharing state between tests (i.E. `func aClusterPool() *hivev1.ClusterPool { return &hivev1.ClusterPool{... }`

Reply:
Done.